src/codegen/sdk/core/import_resolution.py

```diff
@@ -221,6 +221,17 @@ def is_symbol_import(self) -> bool:
         """
         return not self.is_module_import()
 
+    @reader
+    def usage_is_ascertainable(self) -> bool:
```

Comment:
Can we combine this with remove_if_unused and not remove wildcards/side-effects automatically?